compiler/rustc_codegen_llvm/src/intrinsic.rs

```diff
@@ -340,6 +340,42 @@ impl<'ll, 'tcx> IntrinsicCallBuilderMethods<'tcx> for Builder<'_, 'll, 'tcx> {
                     self.const_i32(cache_type),
                 ])
             }
+            sym::carrying_mul_add => {
+                let (size, signed) = fn_args.type_at(0).int_size_and_signed(self.tcx);
+
+                let wide_llty = self.type_ix(size.bits() * 2);
+                let [a, b, c, d] = args.as_array().unwrap().map(|a| {
+                    if signed {
+                        self.sext(a.immediate(), wide_llty)
+                    } else {
+                        self.zext(a.immediate(), wide_llty)
+                    }
```

Comment:
Small nit, but isn't this the same as `self.intcast(a.immediate(), wide_llty, signed)` ?